zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md

```diff
@@ -0,0 +1,388 @@
+
+详见 [文档](https://github.com/ruby/typeprof/blob/master/doc/doc.md) 和 [示例]](https://github.com/ruby/typeprof/blob/master/doc/demo.md)。
+
+TypeProf 目前是实验特性，还不够成熟。只有 Ruby 语言的一个子集可以被分析，能检测出的类型错误也比较有限。但其正在快速开发来覆盖更多 Ruby 语言特性、更好的分析性能以及可用性。我们欢迎任何反馈。
+
+## 其它值得注意的新特性
+
+* 单行模式匹配被重新设计了（实验性）
+
+    * 加入 `=>`。其可以被用作向右赋值。
+
+      ```ruby
+      0 => a
+      p a #=> 0
+
+      {b: 0, c: 1} => {b:}
+      p b #=> 0
+      ```
+    * `in` 改成了会返回 `true` 或 `false`.
+
+      ```ruby
+      # version 3.0
+      0 in 1 #=> false
+
+      # version 2.7
+      0 in 1 #=> raise NoMatchingPatternError
+      ```
+
+* 加入了模式查找功能（实验性）
+
+  ``` ruby
+  case ["a", 1, "b", "c", 2, "d", "e", "f", 3]
+  in [*pre, String => x, String => y, *post]
+    p pre  #=> ["a", 1]
+    p x    #=> "b"
+    p y    #=> "c"
+    p post #=> [2, "d", "e", "f", 3]
+  end
+  ```
+
+* 加入了无尽函数定义
+
+  ``` ruby
+  def square(x) = x * x
+  ```
+
+* 内建了 `Hash#except`
+
+  ``` ruby
+  h = { a: 1, b: 2, c: 3 }
+  p h.except(:a) #=> {:b=>2, :c=>3}
+  ```
+
+* 内存查看以实验性的形式加入
+
+    * 这是一个用来交换内存原始空间的新 C-API。例如将数组和位图在两个扩展库中交换。该扩展库可以互相共享其内存空间的元信息，例如尺寸、格式等。利用这些元信息，扩展库之间可以共享例如多维度的数组。该功能参考了 Python 的 buffer 协议设计。
+
+## 性能提升
+
+* 粘贴长代码到 IRB 中，比起 Ruby 2.7.0 快了 53 倍。例如粘贴 [样例代码](https://gist.github.com/aycabta/30ab96334275bced5796f118c9220b0b) 从 11.7 秒变成了只需 0.22 秒。
+
+
+<video autoplay="autoplay" controls="controls" muted="muted" width="764" height="510" poster="https://cache.ruby-lang.org/pub/media/ruby-3.0-irb-highspeed.png">
+  <source src="https://cache.ruby-lang.org/pub/media/ruby-3.0-irb-highspeed.mp4" type="video/mp4">
+</video>
+
+
+* IRB 加入了 `measure` 命令，来简单测量命令的执行时间。
+
+  ```
+  irb(main):001:0> 3
+  => 3
+  irb(main):002:0> measure
+  TIME is added.
+  => nil
+  irb(main):003:0> 3
+  processing time: 0.000058s
+  => 3
+  irb(main):004:0> measure :off
+  => nil
+  irb(main):005:0> 3
+  => 3
+  ```
+
+## 其它自 2.7 起的新特性
+
+* 关键字参数和其它参数分离。
+  * 原则上，在 2.7 会跑出警告的操作在 3.0 不能执行。详见 [文档](https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/)。
+  * 顺带一提，参数现在支持了前缀参数匹配。
+
+    ``` ruby
+    def method_missing(meth, ...)
+      send(:"do_#{ meth }", ...)
+    end
+    ```
+
+* 模式匹配 (`case/in`) 不再是实验性特性。
+* `$SAFE` 特性被彻底移除，现在它就是一个普通的全部常量。
+* bactrace 的顺序在 Ruby 2.5 中被颠倒，现在倒了回来。现在其行为和 2.4 一致，先打印错误信息，再以此打印其调用者。
+* 一些标准库被升级
+  * RubyGems 3.2.3
+  * Bundler 2.2.3
+  * IRB 1.2.6
+  * Reline 0.1.5
+  * Pysch 3.3.0
+  * JSON 2.5.0
+  * BigDecimal 3.0.0
+  * CSV 3.1.9
+  * Date 3.1.1
+  * Digest 3.0.0
+  * Fiddle 1.0.5
+  * StringIO 3.0.0
+  * StringScanner 3.0.0
+  * etc.
+* 标准库以及内置的 gem 不再含有下面的 gems
+  如果要使用相应的特性，请安装对应的 gem
+  * sdbm
+  * webrick
+  * net-telnet
+  * xmlrpc
+* 下面的 gem 被默认内置
```

Comment:
_Bundled Gems_ are different from _Default Gems_, see https://stdgems.org/#about-the-gemified-standard-library

```suggestion
* 以下 gems 将从默认变为内置
```